drivers/bluetooth/hci/userchan.c

```diff
@@ -99,9 +99,10 @@ static struct net_buf *get_rx(const uint8_t *buf)
  * @brief Decode the length of an HCI H4 packet
  * @details Decodes packet length according to Bluetooth spec v5.4 Vol 4 Part E
  * @param buf	Pointer to a HCI packet buffer
- * @return Length of the HCI packet in bytes, zero if no valid packet found.
+ * @return Length of the HCI packet in bytes, -1 if no valid packet found, 0 if
```

Comment:
Originally the purpose of this function was to only decode the packet length format. With the new logic introduce, this function will not decode the packet length if the frame is not complete. Perhaps this should be mentioned for mantainance/debugging purposes.